### PR TITLE
Support adding custom headers

### PR DESCRIPTION
**Pull request by ptv999duke**
Adds the ability to specify custom email headers based on available fields of the event.

### CODE REVIEW

**Comment by ptv999duke** [comment]
This change is the same as the original PR #56 I submitted before. However, I closed original PR #56 because I realized that I had made one of my commits with the wrong user email, and I would prefer that all the commits be shown as coming from one author.

**Comment by jstoja** [comment]
Hello @ptv999duke Do you have an example of for what this could be used for?

**Comment by ptv999duke** [comment]
Our use case is that we need to attach unique ids to all the internal emails we send via logstash, for tracking/filing purposes. However, we've gotten requests that the unique id not be in a naturally visible location for the email. The best location we found was adding custom headers to the email.

**Comment by aramperes** [comment]
Any chance this can be merged in?

**Comment by jsvd** [comment]
@logstashmachine bump minor

File: docs/index.asciidoc [hunk 1]
```diff
@@ -165,6 +166,14 @@ actual hostname of the connecting client.
 
 The fully-qualified email address for the From: field in the email.
 
+[id="plugins-{type}s-{plugin}-headers"]
+===== `headers`
```
Comment (aramperes) [inline docs/index.asciidoc:170]: It is defined as `custom_headers` in the code

Comment (jsvd) [reply docs/index.asciidoc]: thanks for spotting!

**Pull request closed without merging**